src/Logger.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Krokedil\Support;
+
+use WC_Log_Levels;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Logger.
+ */
+class Logger {
+
+	/**
+	 * WC logger.
+	 *
+	 * @var \WC_Logger
+	 */
+	private $logger;
+
+	/**
+	 * Gateway ID.
+	 *
+	 * @var string
+	 */
+	private $id;
+
+	/**
+	 * Whether logging is enabled.
+	 *
+	 * @var bool
+	 */
+	private $enabled = false;
+
+	/**
+	 * Whether extended logging is enabled.
+	 *
+	 * If enabled, WC_Log_Levels::DEBUG messages will be logged.
+	 *
+	 * @var bool
+	 */
+	private $extended_logging = false;
+
+	/**
+	 * Logger constructor.
+	 *
+	 * @param string $id Gateway ID.
+	 */
+	public function __construct( $id ) {
+		$this->logger = new \WC_Logger();
+		$this->id     = $id;
+
+		add_filter( "woocommerce_settings_api_form_fields_{$this->id}", array( $this, 'add_settings_fields' ) );
+
+		$settings               = get_option( 'woocommerce_' . $this->id . '_settings', array() );
+		$this->enabled          = isset( $settings['logging'] ) && wc_string_to_bool( $settings['logging'] );
+		$this->extended_logging = isset( $settings['extended_logging'] ) && wc_string_to_bool( $settings['extended_logging'] );
```

Comment:
This should be an optional thing to add in, and not just be included by default. We have some plugins that already handles this or has the placement of these fields in a specific place.

This also can not be used with for example shipping method instance settings either in this case. We can discuss on how to do this a bit if you want.